src/domain/buildingBlocks/Pluto.ts

```diff
@@ -1,50 +1,51 @@
-import { DID } from "../models";
+import { CredentialMetadata, DID, LinkSecret, PrismDID } from "../models";
 import { DIDPair } from "../models/DIDPair";
 import { PrivateKey } from "../models";
 import { Mediator } from "../models/Mediator";
 import { Message } from "../models/Message";
-import { PrismDIDInfo } from "../models/PrismDIDInfo";
 import { Credential } from "../models/Credential";
-import { Anoncreds } from "../models/Anoncreds";
 import { PeerDID } from "../../peer-did/PeerDID";
 
+export namespace Pluto {
+  /**
+   * Storable
+   * define properties a Domain object must implement to be compatible with Pluto
+   */
+  export interface Storable {
+    /**
+     * Universally Unique Identifier.
+     * should be unique across all items.
+     */
+    uuid?: string;
```

Comment:
uuid is optional here because a domain object might not have been stored, so a uuid wouldn't be set.
we could pivot to creating the uuid here, instead of relying on the store, it would simplify a lot of the logic, but comes with some risks etc?
@milosh86 what are your thoughts?

Reply:
@curtis-h I get now how it works (`store.insert` generates the `uuid`) ... I see 2 options which could simplify it a bit:

1. whatever implements Pluto.Storable generates `uuid` in its constructor, so we move `uuid` to be required. Then `store` just uses that value instead of worrying about how to generate and parse `uuid`.
2. we use similar approach as in the backend with DTOs and Entities. So, input to `storeXYZ` functions is DTO (no `uuid`) and response is an Entity (with `uuid`). I think issue here is that we use the same abstraction (Domain classes) for both.

Reply:
try option 1

Reply:
uuid now being generated in domain instances